From 7a6712f468650029ee7c10664c4ba02a4d7909b9 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Fri, 17 Jul 2020 21:22:03 +0300
Subject: [PATCH 5/5] mem: Test only the first bank

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 cmd/mem.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/cmd/mem.c b/cmd/mem.c
index 6d54f19527..a507ea3128 100644
--- a/cmd/mem.c
+++ b/cmd/mem.c
@@ -881,8 +881,10 @@ static int do_mem_mtest(cmd_tbl_t *cmdtp, int flag, int argc,
 	const int alt_test = 0;
 #endif
 
-	start = CONFIG_SYS_MEMTEST_START;
-	end = CONFIG_SYS_MEMTEST_END;
+	/* Test only the first bank */
+	start = gd->bd->bi_dram[0].start;
+	end = start + gd->bd->bi_dram[0].size *
+		MEMTEST_NUMERATOR / MEMTEST_DIVIDER;
 
 	if (argc > 1)
 		if (strict_strtoul(argv[1], 16, &start) < 0)
-- 
2.11.0

