From 909503201da187681e9c6c54a8b6dd3643e5490d Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Wed, 8 Jun 2022 11:44:59 +0300
Subject: [PATCH] net: enable random mac if the som eeprom is empty

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 board/compulab/common/eeprom.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/board/compulab/common/eeprom.c b/board/compulab/common/eeprom.c
index defd9b6446..68908f0e34 100644
--- a/board/compulab/common/eeprom.c
+++ b/board/compulab/common/eeprom.c
@@ -15,6 +15,7 @@
 #include <eeprom_field.h>
 #include <linux/kernel.h>
 #include <asm/setup.h>
+#include <net.h>
 #include "eeprom.h"
 
 #ifndef CONFIG_SYS_I2C_EEPROM_ADDR
@@ -151,7 +152,15 @@ int cl_eeprom_read_mac_addr(uchar *buf, uint eeprom_bus)
 	offset = (cl_eeprom_layout != LAYOUT_LEGACY) ?
 			MAC_ADDR_OFFSET : MAC_ADDR_OFFSET_LEGACY;
 
-	return cl_eeprom_read(offset, buf, 6);
+	err = cl_eeprom_read(offset, buf, 6);
+	{ /* generate a random address if the som eeprom is empty */
+		u32 mac0, mac2;
+		mac0 = (u32) buf[0];
+		mac2 = (u32) buf[2];
+		if ( mac0 == mac2 )
+			net_random_ethaddr(buf);
+	}
+	return err;
 }
 
 static u32 board_rev;
-- 
2.17.1

